### DISCUSSION TITLE
Setting name_overwrite=True causes permissions error

### DISCUSSION THREAD

**Discussion by edcloudcycle** [kevincar/bless#163] [Q&A]
Hi all,
Just trying out the library to see if I can use it to simulate one of our devices for testing. To do that it needs to pretend to be that device by having the same name, however setting `name_overwrite=True` causes a `PermissionError: [WinError 5] Access is denied`.

I can see there are a lot of other services available on the BLE, many from Windows providing mobile phone connection options. I guess those services don't want me changing the name. Maybe they need to be disabled or maybe I am better of with a USB BLE dongle that won't have anything else trying to use it.

All suggestions very welcome.

Thanks

Ed

**Comment by edcloudcycle** [comment]
If I turn off Bluetooth in Windows I can see there is a radio_not_present error inside the service provider object so I need a more subtle tool. :)

**Comment by kevincar** [comment]
Converting this to a discussion because it is not yet clear that this is an issue.

**Comment by kevincar** [comment]
Each operating system does this differently. Linux is the most flexible, MacOS is the most strict and does not allow this, and Windows is in between. 

Under the hood, what is really happening is that Bless attempts to change a registry value within the Windows Registry that defines the name of the Bluetooth adapter. You can see this [here](https://github.com/kevincar/bless/blob/73d4b21e461f44b9de4ece66c5a6aeb977f87978/bless/backends/winrt/ble/adapter.py#L31).

Thus, Python may be experiencing a permissions issue when writing to the registry. 

Regarding BLE dongles outside of the on-board one provided with the system, the WinRT BluetoothLE GATT API doesn't seem to allow developers to specify the specific Bluetooth adapter that the GATT services should be broadcasting from.
